### PR TITLE
Migrate to mix release

### PR DESCRIPTION
**Pull request by kamilwaz**
Migrate releases from `distillery` to `mix release`.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -39,9 +40,7 @@ defmodule MongoosePush.Mixfile do
       {:plug_cowboy, "~> 2.2"},
       {:jason, "~> 1.4"},
       {:poison, "~> 3.0", override: true},
-      {:distillery, "~> 2.0", override: true},
       {:confex, "~> 3.2", override: true},
-      {:mix_docker, "~> 0.5"},
```
Comment (michalwski) [inline mix.exs:44]: Why the `mix_docker` was removed?

**Review by michalwski — COMMENTED** [review]
I had only one question.

**Comment by michalwski** [comment]
One more thing. Could you point to a CircleCi log showing that MongoosPush starts in docker and we can talk to it like we did before?

File: mix.exs (hunk 1 above)
Comment (kamilwaz) [reply mix.exs:44]: I removed it because it wasn't used anywhere at all. Additionally, this library is no longer maintained.

**Comment by kamilwaz** [comment]
> One more thing. Could you point to a CircleCi log showing that MongoosPush starts in docker and we can talk to it like we did before?

Here: https://app.circleci.com/pipelines/github/esl/MongoosePush/1191/workflows/9af95ef1-db85-4343-829f-c25532340608/jobs/9482?invite=true#step-111-2. 

`mix test.env.wait` waits for all components to be up and running, including MongoosePush (https/8443).

**Review by michalwski — APPROVED** [review]
Approving

**Pull request merged**